### PR TITLE
Feature/saas 12636 azure ensure waf policy for application gateway is set to prevention mode

### PR DESCRIPTION
**Pull request by abdullahaslam306**

### CODE REVIEW

File: plugins/azure/applicationGateway/agPreventionModeEnabled.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Application Gateway WAF Prevention Mode Enabled',
+    category: 'Application Gateway',
+    domain: 'Application Gateway',
+    description: 'Ensure that WAF policy for Microsoft Azure Application gateway is set to Prevention mode',
```
Comment (AkhtarAmir) [inline plugins/azure/applicationGateway/agPreventionModeEnabled.js:8]:
```suggestion
    description: 'Ensure that WAF policy for Microsoft Azure Application gateway is set to Prevention mode.',
```

File: plugins/azure/applicationGateway/agPreventionModeEnabled.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                return rcb();
+            }
+
+            if (!appGateways.data.length) {
+                helpers.addResult(results, 0, 'No existing Application Gateway found', location);
+                return rcb();
+            } 
+            
+            for (let appGateway of appGateways.data) {
+                if (!appGateway.id || !appGateway.sku.tier === 'WAF_v2') continue;
```
Comment (AkhtarAmir) [inline plugins/azure/applicationGateway/agPreventionModeEnabled.js:36]: Is it not supported for v2? If not, then add a result saying 'Prevention mode is not supported for WAF_v2'

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/azure/applicationGateway/agPreventionModeEnabled.js (hunk 2 above)
Comment (abdullahaslam306) [reply plugins/azure/applicationGateway/agPreventionModeEnabled.js]: It is only available for WAFV2 I have added message in case of other standard

**Review by AkhtarAmir — APPROVED** [review]

File: plugins/azure/applicationGateway/agPreventionModeEnabled.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            
+            for (let appGateway of appGateways.data) {
+                if (!appGateway.id) continue;
+                
+                if (appGateway.sku.tier != 'WAF_v2'){
+                    helpers.addResult(results, 2, 'Prevention mode is not supported for WAF Standard v2 tier', location, appGateway.id);
+                    continue;
+                }
+
+                if (appGateway.webApplicationFirewallConfiguration && appGateway.webApplicationFirewallConfiguration.firewallMode.toLowerCase() === 'prevention') {
```
Comment (AkhtarAmir) [inline plugins/azure/applicationGateway/agPreventionModeEnabled.js:43]:
```suggestion
                if (appGateway.webApplicationFirewallConfiguration && appGateway.webApplicationFirewallConfiguration.firewallMode && appGateway.webApplicationFirewallConfiguration.firewallMode.toLowerCase() === 'prevention') {
```

**Pull request merged**